### PR TITLE
[camera] improve camera API for java

### LINKED ISSUE #303: Usage of camera animation plugin
## New Feature

I tried my best to use the camera animation plugin on beta build 18, but when I'm doing MapboxMap.flyTo the function doesn't seem to exists. I tried multiple things, didn't find any animation functions (either from MapboxMap or Mapview).

I'm developing in Java.
I added the camera dependencies in the graddle.build and that's all. Am I missing something?

So it would be nice to update the documentation.
Thank you!

## Why
Cause everyone may have this problem

### PR DESCRIPTION
**Pull request by tobrun**
Closes #303 

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Improve camera API consumption from java programming language by adding  a JvmName annotations.</changelog>`.

### Summary of changes

This PR moves the camera extension functions to a dedicated file and add a JvmName to it. Also renames the camera getter to be optimized for consumption through the java programming language. 

### User impact (optional)

Users will be able to use the camera animations plugin more easily.

### CODE REVIEW

**Review by Chaoba — APPROVED** [review]

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsExt.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.ScreenCoordinate
+import com.mapbox.maps.plugin.delegates.MapPluginExtensionsDelegate
+import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
+
+/**
+ * Extension val for MapView to get the Camera Animations plugin instance.
+ */
+val MapPluginProviderDelegate.camera: CameraAnimationsPlugin
+  @JvmName("getCameraAnimationsPlugin")
```
Comment (kiryldz) [inline plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsExt.kt:13]: why so long naming?
perhaps just `camera()`?

File: app/src/main/java/com/mapbox/maps/testapp/examples/JavaInterfaceTest.java [hunk 2]
```diff
@@ -28,4 +35,27 @@ private void addSymbolLayer(Style style) {
         symbolLayer.iconOpacity(expression);
         symbolLayer.iconColor(rgba(255.0, 0.0, 0.0, 1.0));
     }
+
+    private void cameraAnimations(MapView mapView) {
+        MapboxMap mapboxMap = mapView.getMapboxMap();
```
Comment (kiryldz) [inline app/src/main/java/com/mapbox/maps/testapp/examples/JavaInterfaceTest.java:40]: nit here and below - almost all could be `final` 😄

**Review by kiryldz — APPROVED** [review]
LGTM with couple nit's

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsExt.kt (hunk 1 above)
Comment (tobrun) [reply plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsExt.kt:13]: I'm calling it as you named it :) I'm fine with `getCamera()`

**Pull request merged**